File: app/src/main/java/me/jiahuan/androidlearn/MainActivity.java
package me.jiahuan.androidlearn;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.util.Log;
import android.view.MenuItem;

import java.util.List;

import me.jiahuan.androidlearn.base.BaseActivity;
import me.jiahuan.androidlearn.function.FunctionFragment;
import me.jiahuan.androidlearn.ui.UIFragment;

public class MainActivity extends BaseActivity {

    private static final String TAG = "MainActivity";

    BottomNavigationView mBottomNavigationView;

    private FunctionFragment mFunctionFragment;
    private UIFragment mUIFragment;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d(TAG, "onCreate");
        initializeActivity(R.layout.layout_main_activity);
        initialize();
    }

    private void initialize() {
        mBottomNavigationView = findViewById(R.id.id_layout_main_activity_bottom_navigation_view);
        mFunctionFragment = new FunctionFragment();
        mUIFragment = new UIFragment();
        mBottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                addOrShowFragment(item.getItemId());
                return true;
            }
        });
        addOrShowFragment(R.id.id_menu_main_activity_bottom_navigation_view_function);
    }


    private void hideAllFragments() {
        FragmentManager fragmentManager = getSupportFragmentManager();
        List<Fragment> fragments = fragmentManager.getFragments();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        for (Fragment fragment : fragments) {
            fragmentTransaction.hide(fragment);
        }
        fragmentTransaction.commit();
    }

    private void addOrShowFragment(int itemId) {
        hideAllFragments();
        FragmentManager fragmentManager = getSupportFragmentManager();
        List<Fragment> fragments = fragmentManager.getFragments();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        if (itemId == R.id.id_menu_main_activity_bottom_navigation_view_function) {
            if (fragments.contains(mFunctionFragment)) {
                fragmentTransaction.show(mFunctionFragment);
            } else {
                fragmentTransaction.add(R.id.id_layout_main_activity_fragment_container_view_group, mFunctionFragment, FunctionFragment.TAG);
            }
            getSupportActionBar().setTitle("功能");
        } else if (itemId == R.id.id_menu_main_activity_bottom_navigation_view_ui) {
            if (fragments.contains(mUIFragment)) {
                fragmentTransaction.show(mUIFragment);
            } else {
                fragmentTransaction.add(R.id.id_layout_main_activity_fragment_container_view_group, mUIFragment, UIFragment.TAG);
            }
            getSupportActionBar().setTitle("UI");
        }
        fragmentTransaction.commit();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        Log.d(TAG, "onSaveInstanceState");
    }


    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG, "onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "onStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "onDestroy");
    }
}

File: app/src/main/kotlin/me/jiahuan/androidlearn/MainActivity.kt
package me.jiahuan.androidlearn

import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.os.PersistableBundle
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.util.Log
import kotlinx.android.synthetic.main.main_layout_main_activity.*
import me.jiahuan.androidlearn.handler.HandlerActivity
import me.jiahuan.androidlearn.propertyanimation.PropertyAnimationActivity
import me.jiahuan.androidlearn.pullrefresh.PullRefreshActivity
import me.jiahuan.androidlearn.react.ReactNativeActivity
import me.jiahuan.androidlearn.service.ServiceActivity
import me.jiahuan.androidlearn.skin.SkinChangeActivity
import me.jiahuan.androidlearn.threadpool.ThreadPoolActivity


class MainActivity : AppCompatActivity() {

    companion object {
        private val TAG = "MainActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG, "onCreate, savedInstanceState = " + savedInstanceState)
        setContentView(R.layout.main_layout_main_activity)
        initialize()
        activity_main_pull_refresh_text_view.setOnClickListener {
            val intent = Intent(this, PullRefreshActivity::class.java)
            startActivity(intent)
        }
        activity_main_property_animation_text_view.setOnClickListener {
            val intent = Intent(this, PropertyAnimationActivity::class.java)
            startActivity(intent)
        }
        activity_main_bind_service_text_view.setOnClickListener {
            startActivity(Intent(this, ServiceActivity::class.java))
        }
        activity_main_open_ta_button.setOnClickListener {
            startActivity(Intent(this, TransparentActivity::class.java))
        }
        activity_main_open_dialog_button.setOnClickListener {
            AlertDialog.Builder(this@MainActivity).setMessage("XXXX").show()
        }
        activity_main_handler_activity_button.setOnClickListener {
            startActivity(Intent(this@MainActivity, HandlerActivity::class.java))
        }
        activity_main_react_native_button.setOnClickListener {
            startActivity(Intent(this, ReactNativeActivity::class.java))
        }
        activity_main_change_skin_button.setOnClickListener {
            startActivity(Intent(this, SkinChangeActivity::class.java))
        }
        activity_main_thread_pool_button.setOnClickListener {
            startActivity(Intent(this, ThreadPoolActivity::class.java))
        }
    }


    private fun initialize() {
        setSupportActionBar(main_layout_main_activity_tool_bar)
        if (supportActionBar != null) {
            supportActionBar!!.title = "测试"
        }

        val actionBarDrawerToggle = ActionBarDrawerToggle(this@MainActivity, main_layout_main_activity_drawer_layout, main_layout_main_activity_tool_bar, R.string.main_string_main_activity_navigation_drawer_open, R.string.main_string_main_activity_navigation_drawer_close)
        actionBarDrawerToggle.syncState()
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "onPause")
    }


    override fun onResume() {
        super.onResume()
        Log.d(TAG, "onResume")
    }


    override fun onStop() {
        super.onStop()
        Log.d(TAG, "onStop")
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "onStart")
    }


    override fun onConfigurationChanged(newConfig: Configuration?) {
        super.onConfigurationChanged(newConfig)
        Log.d(TAG, "onConfigurationChanged")
    }

    override fun onSaveInstanceState(outState: Bundle?, outPersistentState: PersistableBundle?) {
        super.onSaveInstanceState(outState, outPersistentState)
        Log.d(TAG, "onSaveInstanceState new")
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        Log.d(TAG, "onSaveInstanceState")
    }
}
